docs/guides/upgrading-v3.md

```diff
@@ -0,0 +1,265 @@
+
+# Correct syntax:
+serverless deploy --verbose
+```
+
+This change makes the CLI much more robust at detecting arguments from options and their values.
+
+On that note, the `-v` option is no longer recognized (it was ambiguous with `--version`): use the full `--verbose` option instead.
+
+When the `serverless` CLI is installed globally and locally (in the project’s `node_modules`), the local version will always be used. It is no longer possible to disable that behavior ([learn more](../deprecations.md#support-for-enablelocalinstallationfallback-setting-is-to-be-removed)).
+
+Finally, the `serverless studio` command has been removed: that feature was deprecated and is no longer available.
+
+### Service configuration
+
+The default Lambda runtime has changed from NodeJS 12 to NodeJS 14, given this is now the default runtime recommended by AWS.
+
+In `serverless.yml`, the `service` key no longer accepts a YAML object ([learn more](../deprecations.md#service-property-object-notation)).
+
+```yaml
+# Deprecated in v2, removed in v3:
+service:
+    name: my-service
+
+# Correct syntax:
+service: my-service
+```
+
+All options that used to be defined inside the `service` key have been moved to other sections (mentioned below in that document). This change clears up confusion that existed between the `service` and `provider` sections.
+
+CloudFormation extensions defined in `resources.extensions` are now validated: extensions that point to unknown resources will throw an error, which should help catch broken configuration before deploying ([learn more](../deprecations.md#defining-extensions-to-nonexistent-resources-in-resourcesextensions)).
+
+### API Gateway
+
+When configuring API Gateway, some configuration options have moved to a dedicated sub-section of `provider`. That will help clear up confusion with similar `httpApi` settings.
+
+```yaml
+provider:
+  # Deprecated in v2, removed in v3:
+  apiKeys: ...
+  resourcePolicy: ...
+  usagePlan: ...
+
+  # Correct syntax:
+  apiGateway:
+    apiKeys: ...
+    resourcePolicy: ...
+    usagePlan: ...
+```
+
+The `schema` option on HTTP events has also been renamed to `schemas`. That change allowed us to support much more schema validation features:
+
+```yaml
+functions:
+  hello:
+    handler: hello.handler
+    events:
+      - http:
+        ...
+        request:
+          # Deprecated in v2, removed in v3:
+          schema: ...
+          # Correct syntax:
+          schemas: ...
+```
+
+Learn more about [schema changes](../deprecations.md#aws-api-gateway-schemas).
+
+When an external API Gateway is used and imported via `provider.apiGateway.restApiId`, both `provider.logs.restApi` and `provider.tracing.apiGateway` options will be ignored. These settings are applicable only if the API Gateway is provisioned by Serverless Framework.
+
+The CloudFormation tags defined in `provider.tags` will now be correctly applied to HTTP APIs ([learn more](../deprecations.md#http-api-provider-tags)).
```

Comment:
Maybe let's add the word `stage` at the end? Because the tags are applied to HTTP API stage only